sdk/monitor/azure-monitor-ingestion/tests/perf_tests/upload_logs.py

```diff
@@ -0,0 +1,66 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+from datetime import datetime
+
+from azure_devtools.perfstress_tests import PerfStressTest
+from azure.identity import DefaultAzureCredential
+from azure.identity.aio import DefaultAzureCredential as AsyncDefaultAzureCredential
+from azure.monitor.ingestion import LogsIngestionClient
+from azure.monitor.ingestion.aio import LogsIngestionClient as AsyncLogsIngestionClient
+
+
+class UploadLogsTest(PerfStressTest):
+
+    def __init__(self, arguments):
+        super().__init__(arguments)
+
+        self.data_collection_rule_id = self.get_from_env("AZURE_MONITOR_DCR_ID")
+        self.data_collection_endpoint = self.get_from_env("AZURE_MONITOR_DCE")
+        self.stream_name = self.get_from_env("AZURE_MONITOR_STREAM_NAME")
+
+        # Create clients
+        self.client = LogsIngestionClient(
+            endpoint=self.data_collection_endpoint,
+            credential=DefaultAzureCredential()
+        )
+        self.async_client = AsyncLogsIngestionClient(
+            endpoint=self.data_collection_endpoint,
+            credential=AsyncDefaultAzureCredential()
+        )
+
+        # Create log entries to upload
+        # TODO: Parameterize the size of each log entry
+        self.logs = []
+        for i in range(self.args.num_logs):
+            self.logs.append({
+                "Time": datetime.now().isoformat(),
+                "Computer": f"Computer {i}",
+                "AdditionalContext": "some additional context"
+            })
```

Comment:
I would move this to the setup function. Just in case someone decided to test with many 1000s of logs.

Reply:
Thanks. Moved the log generation to the setup method.